src/vmm/src/resources.rs

```diff
@@ -241,6 +244,34 @@ impl VmResources {
         &self.vm_config
     }
 
+    #[allow(unused_variables)]
+    #[allow(unused_results)]
+    fn validate_cpu_configuration(
+        &mut self,
+        cpu_config: &CpuConfigurationSet,
+    ) -> Result<CpuConfigError> {
+        for cpu_config_entry in &cpu_config.cpu_features {
+            if !CPU_FEATURE_INDEX_MAP.contains_key(cpu_config_entry.name.as_str()) {
+                return Err(CpuConfigError::InvalidCpuFeatureName);
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Configure vCPU features.
+    #[allow(unused_results)]
+    pub fn configure_cpu(&mut self, cpu_config: CpuConfigurationSet) -> Result<CpuConfigError> {
+        // Validate the CPU configuration first
+        let validation_result = self.validate_cpu_configuration(&cpu_config);
+        if validation_result.is_ok() {
+            // Update the CPU configuration on the VM
+            self.vm_config.cpu_template = CUSTOM(cpu_config);
+        }
+
+        validation_result
+    }
```

Comment:
Could use error propagation here like:
```suggestion
    /// Configure vCPU features.
    #[allow(unused_results)]
    pub fn configure_cpu(&mut self, cpu_config: CpuConfigurationSet) -> Result<CpuConfigError> {
        // Validate the CPU configuration first
        let validation = self.validate_cpu_configuration(&cpu_config)?; // Error propagation here.
        // Update the CPU configuration on the VM
        self.vm_config.cpu_template = CUSTOM(cpu_config);
        // Return
        validation 
    }
```

Reply:
I do think this is a cleaner way to go, but does require a factory of sorts for the CUSTOM type which I'm going to defer for now.